src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -1222,4 +1223,1555 @@ struct S3
+    {
+        var c = @"
+public class C
+{
+    public required int Prop1 { get; set; }
+    public required int Field1;
+}
+";
+
+        var creation = @"
+C c = new" + constructor + @"() { Prop1 = 1, Field1 = 1 };
+System.Console.WriteLine($""{c.Prop1}, {c.Field1}"");
+";
+        var comp = CreateCompilationWithRequiredMembers(new[] { c, creation });
+        CompileAndVerify(comp, expectedOutput: "1, 1");
+
+        var cComp = CreateCompilationWithRequiredMembers(c);
+        comp = CreateCompilation(creation, references: new[] { useMetadataReference ? cComp.ToMetadataReference() : cComp.EmitToImageReference() });
+        CompileAndVerify(comp, expectedOutput: "1, 1");
+    }
+
+    [Fact]
+    public void EnforcedRequiredMembers_NoInheritance_Unsettable()
+    {
+        var c = @"
+var c = new C();
+c = new C() { Prop1 = 1, Field1 = 1 };
+
+public class C
+{
+    public required int Prop1 { get; }
+    public required readonly int Field1;
+}
+";
+        var comp = CreateCompilationWithRequiredMembers(c);
+        comp.VerifyDiagnostics(
+            // (2,13): error CS9506: Required member 'C.Prop1' must be set in the object initializer or attribute constructor.
+            // var c = new C();
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "C").WithArguments("C.Prop1").WithLocation(2, 13),
+            // (2,13): error CS9506: Required member 'C.Field1' must be set in the object initializer or attribute constructor.
+            // var c = new C();
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSet, "C").WithArguments("C.Field1").WithLocation(2, 13),
+            // (3,15): error CS0200: Property or indexer 'C.Prop1' cannot be assigned to -- it is read only
+            // c = new C() { Prop1 = 1, Field1 = 1 };
+            Diagnostic(ErrorCode.ERR_AssgReadonlyProp, "Prop1").WithArguments("C.Prop1").WithLocation(3, 15),
+            // (3,26): error CS0191: A readonly field cannot be assigned to (except in a constructor or init-only setter of the type in which the field is defined or a variable initializer)
+            // c = new C() { Prop1 = 1, Field1 = 1 };
+            Diagnostic(ErrorCode.ERR_AssgReadonly, "Field1").WithLocation(3, 26),
+            // (7,25): error CS9505: Required member 'C.Prop1' must be settable.
+            //     public required int Prop1 { get; }
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSettable, "Prop1").WithArguments("C.Prop1").WithLocation(7, 25),
+            // (8,34): error CS9505: Required member 'C.Field1' must be settable.
+            //     public required readonly int Field1;
+            Diagnostic(ErrorCode.ERR_RequiredMemberMustBeSettable, "Field1").WithArguments("C.Field1").WithLocation(8, 34)
+        );
+    }
+
+    [Fact]
+    public void EnforcedRequiredMembers_NoInheritance_DisallowedNestedObjectInitializer()
+    {
+        var c = @"
+var c = new C() { D1 = { NestedProp = 1 }, D2 = { NestedProp = 2 } };
```

Comment:
nit: consider adding the version that works for contrast: `var c = new C() { D1 = new D() { NestedProp = 1 }, D2 = new D() { NestedProp = 2 } };`